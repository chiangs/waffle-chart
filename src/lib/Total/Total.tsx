import React from 'react';

type Props = {
    children: any;
};

const Total: React.FC<Props> = (props: Props) => (
    <h3 className='total' data-testid='total'>
        {props.children}
    </h3>
);

export default Total;
