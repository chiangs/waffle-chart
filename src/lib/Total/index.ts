export { default as Total } from './Total';
