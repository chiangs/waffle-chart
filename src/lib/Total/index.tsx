import React from 'react';

type Props = {
    style: {} | undefined;
    children: any;
};

const COMPONENT_NAME = 'total';

const Total: React.FC<Props> = ({ style = undefined, children }) => (
    <h3 className={COMPONENT_NAME} style={style} data-testid={COMPONENT_NAME}>
        {children}
    </h3>
);

export default Total;
