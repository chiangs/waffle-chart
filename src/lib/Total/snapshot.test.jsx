import React from 'react';
import renderer from 'react-test-renderer';
import { describe, expect, test } from 'vitest';
import Total from './index';

describe('DataDisplay', () => {
    test('DataDisplay component renders correctly', () => {
        const props = {};
        const component = renderer.create(<Total {...props} />);
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});
