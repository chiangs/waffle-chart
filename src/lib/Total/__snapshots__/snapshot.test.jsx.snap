// Vitest Snapshot v1

exports[`DataDisplay > DataDisplay component renders correctly 1`] = `
<h3
  className="total"
  data-testid="total"
/>
`;
